jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/TypeUtil.java

```diff
@@ -140,48 +139,30 @@
         class2Name.put(java.lang.String.class, "java.lang.String");
     }
 
-    private static final HashMap<Class<?>, Method> class2Value = new HashMap<>();
+    private static final HashMap<Class<?>, ValueOfString<?>> class2Value = new HashMap<>();
 
+    @FunctionalInterface
+    private interface ValueOfString<T> {
```

Comment:
Can this just be `Function<String, T>`? It's all internal anyway, no need for another type.

Reply:
good idea; see commit below